cuqi/testproblem/_testproblem.py

```diff
@@ -419,25 +419,40 @@ class Poisson_1D(BayesianProblem):
     Parameters
     ------------
     dim : int
-        size of the grid for the poisson problem
+        | size of the grid for the poisson problem
 
     endpoint : float
-        Location of end-point of grid.
+        | Location of end-point of grid.
     
     source : lambda function
-        Function for source term.
+        | Function for source term.
 
     field_type : str or cuqi.geometry.Geometry
-        Field type of domain.
+        | Field type of domain. The accepted values are:
+        | a Geometry object.
+        | "KL": a :class:`cuqi.geometry.KLExpansion` geometry object will be created and set as a domain geometry.
+        | "KL_Full": a :class:`cuqi.geometry.KLExpansion_Full` geometry object will be created and set as a domain geometry.
+        | "Step": a :class:`cuqi.geometry.StepExpansion` geometry object will be created and set as a domain geometry.
+        | "CustomKL": a :class:`cuqi.geometry.CustomKL` geometry object will be created and set as a domain geometry.
+        | None: a :class:`cuqi.geometry.Continuous1D` geometry object will be created and set as a domain geometry.
 
-    KL_map : lambda function
-        Mapping used to modify field.
+    field_params : dict
+        | Passed as argument `params` to the underlying geometry when field type is `KL` or `KL_Full` or `CustomKL`.
 
-    KL_imap : lambda function
-        Inverse of KL map.
+    map : lambda function
+        | Mapping used to modify field.
+
+    imap : lambda function
+        | Inverse of KL map.
 
     SNR : int
-        Signal-to-noise ratio
+        | Signal-to-noise ratio
+
+    n_steps: int, default 3
+        | Number of steps for the "Step" field.
```

Comment:
I think this parameter should be set via field_params instead. This is how the other geometries/fields get their parameters set.

Reply:
Thank you @nabriis for your timely review! 

I think your suggestion is a very good idea but I see a complication here that might require further updates in `KLExpantion`, `KLExpansionFull` and `CustomKL`

For example: `KLExpansion_Full` initializer (and similarly the other 2 KL geometries) expect that params is a dictionary. 

```python
    def __init__(self, grid, params, axis_labels=['x'],**kwargs):

        super().__init__(grid, axis_labels,**kwargs)

        cl = params['cor_len']
        tau2 = 1./cl/cl
        nu = params["nu"]
```

while for example for `StepExpansion` Geometry, the initializer signature is:

```python
def __init__(self, grid, n_steps=3, fun2par_projection='mean', **kwargs):
```

my initial thought is that params in KL geometries should not be a dictionary and should be args or kwargs, for example 

```python
    def __init__(self, grid, nu, cor_len, axis_labels=['x'],**kwargs):
```

Then in Heat_1D one can pass a dictionary param_fields which will be passed with double asterisks `**param_fields` to the underlying geometry (in this case it will work whether the underlying geometry is StepExpansion or KL). 

We did something similar in, for example, plot_envelope parameters in method Samples.plot_ci 

We can talk about this in the stand-up

Reply:
We discussed in standup to switch the KL stuff to keywords. Hopefully wont take too long

Reply:
Thank you @nabriis  for your feedback! I implemented the change we discussed today. To document the KL geometries parameters, I needed to write the expansion formula which was not straight forward to find. Now the user can look at where the parameters are used in the formula, for example, `cor_len`. 

To write the formulas I referred to a wiki article that discuss the idst (inverse discrete sign transform) https://en.wikipedia.org/wiki/Discrete_sine_transform. But most importantly, I wrote 2 methods in test_geometry test module that implements exactly what KLExpantion and KLExpansion_Full does (the par2fun method) but without using the spicy method `idst` that are used in  `KLExpantion` and `KLExpansion_Full`. The purpose of writing these two methods (that I am also using in tests): `_inverse_sin_discrete_transform_KLFull` and `_inverse_sin_discrete_transform_KL` is that they corresponds to the math described in the KL geometries documentation (so to insure the math is written correctly)

Reply:
Wow! This seems really good. I will have a closer look tomorrow.

Reply:
Great, thanks!